### ISSUE TITLE
[Bug] Apline + Go: gives arch=error and a sed bad regex error

### ISSUE THREAD

**Issue by coolaj86** [webinstall/webi-installers#728]
- [x] arch is being detected wrong, but only on part of the process
- [x] busybox sed doesn't like one of our matchers

Maybe because it's a native musl build?

## System

```sh
cat /etc/issue
uname -srm
```

```text
Welcome to Alpine Linux 3.17
Kernel \r on an \m (\l)
```

```text
Linux 5.15.107-2-pve x86_64
```

## Error

```sh
curl https://webi.sh/go | sh
```

```text
Error: no 'go' release for 'linux' on 'amd64' as one of 'tar,exe,zip,xz,git,dmg' by the tag 'stable'
       'go' is available for 'aix,macos,freebsd,linux,windows' on 'ppc64,amd64,arm64,x86,armv6l,ppc64le,s390x' as one of 'tar'
       (check that the package name and version are correct)

sed: bad regex '\?.*': Repetition not preceded by valid expression
```

## Hints

Produces a bad `WEBI_RELEASES` url with `arch=error`:

```sh
WEBI_RELEASES='https://webinstall.dev/api/releases/go@stable.tab?os=linux&arch=error&formats=tar&pretty=true'
```

**Comment by coolaj86** [comment]
The same is true for `pathman`:

```text
Thanks for using webi to install 'pathman' on 'Linux/x86_64'.
Have a problem? Experience a bug? Please let us know:
        https://github.com/webinstall/webi-installers/issues

Lovin' it? Say thanks with a Star on GitHub:
        https://github.com/webinstall/webi-installers

Error: no 'pathman' release for 'linux' on 'amd64' as one of 'tar,exe,zip,xz,git,dmg' by the tag 'stable'
       'pathman' is available for 'macos,linux,windows' on 'arm64,amd64,armv6l,armv7l' as one of 'tar'
       (check that the package name and version are correct)

sed: bad regex '\?.*': Repetition not preceded by valid expression
```

**Comment by coolaj86** [comment]
Fixed in #731
